pkg/infrastructure/services/configmap.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package services
+
+import (
+	"crypto/md5"
+	"fmt"
+	"github.com/RHsyseng/operator-utils/pkg/resource"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"reflect"
+)
+
+const (
+	appPropConfigMapSuffix = "-app-prop"
+	appPropContentKey      = "application.properties"
+	defaultAppPropContent  = ""
+)
+
+// GetAppPropConfigMapContentHash calculates the hash of the application.properties contents in the ConfigMap
+// If the ConfigMap doesn't exist, create a new one and return it.
+func GetAppPropConfigMapContentHash(name, namespace string, cli *client.Client) (string, *corev1.ConfigMap, error) {
+	configMapName := GetAppPropConfigMapName(name)
+	configMap := &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: configMapName, Namespace: namespace}}
+
+	exist, err := kubernetes.ResourceC(cli).Fetch(configMap)
+	if err != nil {
+		return "", nil, err
+	}
+
+	if exist {
+		if _, ok := configMap.Data[appPropContentKey]; !ok {
+			exist = false
+		}
+	}
+
+	if !exist {
+		configMap.Data = map[string]string{
+			appPropContentKey: defaultAppPropContent,
+		}
+	}
+
+	contentHash := fmt.Sprintf("%x", md5.Sum([]byte(configMap.Data[appPropContentKey])))
+
+	if exist {
+		return contentHash, nil, nil
+	}
+
+	return contentHash, configMap, nil
+}
+
+// GetAppPropConfigMapName generates the name of the config map for application.properties
+func GetAppPropConfigMapName(name string) string {
+	return name + appPropConfigMapSuffix
+}
+
+// ExcludeAppPropConfigMapFromResource excludes the applicatio properties config map from the resources
```

Comment:
```suggestion
// ExcludeAppPropConfigMapFromResource excludes the application properties config map from the resources
```

Reply:
Please also consider expanding this doc to explain that this function removes the configMap from kubernetes resources that need to be compared, ignoring the ConfigMap. I took sometime to figure that out. xD